glue/core/component.py

```diff
@@ -549,3 +557,98 @@ def categorical(self):
     @property
     def datetime(self):
         return False
+
+
+class ExtendedComponent(Component):
+    """
+    A data component representing an extent or a region.
+
+    This component can be used when a dataset describes regions or ranges
+    and is typically used with a :class:`~glue.core.data.RegionData` object.
```

Comment:
I have moved some of the key logic around displaying regions from the Viewer (where it would have to be repeated) into the RegionData object. This docstring now mentions why to use RegionData.